// @scaleflex/ui/snackbar/snackbar-context.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import { createContext } from 'react';
import { Vertical, Horizontal } from '@scaleflex/ui/core/popup/types';
// import { MessageVariant } from './types';

// Default props which can be redefined in <SnackbarProvider /> (src/index.js)
export var defaultProps = {
  autoHideDuration: 4000,
  anchorOrigin: {
    vertical: Vertical.Bottom,
    horizontal: Horizontal.Left
  }
};
var SnackbarContext = /*#__PURE__*/createContext(_objectSpread(_objectSpread({}, defaultProps), {}, {
  showMessage: function showMessage() {
    /** will be redefined in provider */
  }
}));
export default SnackbarContext;
