// @scaleflex/ui/core/pagination/pagination.component.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
var _excluded = ["getItemAriaLabel"],
  _excluded2 = ["disabled", "page", "selected", "type"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from 'react';
import PT from 'prop-types';
import Arrow from '../arrow';
import { intrinsicComponent } from '../../utils/functions';
import { usePagination } from '../../hooks/use-pagination';
import Styled from './pagination.styles';
function defaultGetAriaLabel(type, page, selected) {
  if (type === 'page') {
    return "".concat(selected ? '' : 'Go to ', "page ").concat(page);
  }
  return "Go to ".concat(type, " page");
}
var Pagination = intrinsicComponent(function (props, ref) {
  var _usePagination = usePagination(_objectSpread({}, props)),
    items = _usePagination.items;
  var _props$getItemAriaLab = props.getItemAriaLabel,
    getItemAriaLabel = _props$getItemAriaLab === void 0 ? defaultGetAriaLabel : _props$getItemAriaLab,
    rest = _objectWithoutProperties(props, _excluded);
  var paginationItem = function paginationItem(paginationitems) {
    var _paginationitems$disa = paginationitems.disabled,
      disabled = _paginationitems$disa === void 0 ? false : _paginationitems$disa,
      page = paginationitems.page,
      _paginationitems$sele = paginationitems.selected,
      selected = _paginationitems$sele === void 0 ? false : _paginationitems$sele,
      _paginationitems$type = paginationitems.type,
      type = _paginationitems$type === void 0 ? 'page' : _paginationitems$type,
      other = _objectWithoutProperties(paginationitems, _excluded2);
    var Icons = {
      previous: function previous() {
        return /*#__PURE__*/React.createElement(Arrow, {
          type: "left",
          IconProps: {
            size: 8
          }
        });
      },
      next: function next() {
        return /*#__PURE__*/React.createElement(Arrow, {
          type: "right",
          IconProps: {
            size: 8
          }
        });
      }
      // first: FirstPageIcon,
      // last: LastPageIcon,
    };
    var Icon = Icons[type];
    return type === 'start-ellipsis' || type === 'end-ellipsis' ? /*#__PURE__*/React.createElement(Styled.PaginationItem, _extends({
      disabled: disabled,
      selected: selected
    }, other), "\u2026") : /*#__PURE__*/React.createElement(Styled.PaginationItem, _extends({
      ref: ref,
      disabled: disabled,
      selected: selected
    }, other), type === 'page' && page, Icon ? /*#__PURE__*/React.createElement(Icon, null) : null);
  };
  return /*#__PURE__*/React.createElement(Styled.Pagination, {
    ref: ref,
    style: rest.style
  }, /*#__PURE__*/React.createElement(Styled.PaginationList, null, items.map(function (item, index) {
    return /*#__PURE__*/React.createElement(Styled.PaginationItemList, {
      key: index
    }, paginationItem(_objectSpread(_objectSpread({}, item), {}, {
      'aria-label': getItemAriaLabel(item.type, item.page, item.selected),
      varaint: 'text'
    })));
  })));
});
Pagination.defaultProps = {};
export var propTypes = {
  boundaryCount: PT.number,
  count: PT.number,
  defaultPage: PT.number,
  disabled: PT.bool,
  onChange: PT.func,
  getItemAriaLabel: PT.func,
  page: PT.number,
  siblingCount: PT.number
};
Pagination.propTypes = propTypes;
export default Pagination;
