// @scaleflex/ui/core/search/search.component.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["onChange", "value", "size"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
import React from 'react';
import SearchIcon from '@scaleflex/icons/search';
import { intrinsicComponent } from '../../utils/functions';
import { InputSize } from '../../utils/types';
import CrossButton from '../cross-button/cross-button.component';
import Input from '../input';
import Styled from './search.styles';
import { defaultProps as inputDefaultProps, propTypes as inputPropTypes } from '../input/input.component';
var Search = intrinsicComponent(function (_ref, ref) {
  var onChange = _ref.onChange,
    _ref$value = _ref.value,
    value = _ref$value === void 0 ? '' : _ref$value,
    size = _ref.size,
    rest = _objectWithoutProperties(_ref, _excluded);
  var handleClearIconClick = function handleClearIconClick(event) {
    if (onChange) {
      onChange(event);
    }
  };
  var getSearchIconSize = function getSearchIconSize(sizeName) {
    switch (sizeName) {
      case InputSize.Sm:
        return 16;
      case InputSize.Md:
      default:
        return 18;
    }
  };
  return /*#__PURE__*/React.createElement(Styled.Search, {
    size: size
  }, /*#__PURE__*/React.createElement(Input, _extends({
    ref: ref,
    value: value,
    size: size
  }, rest, {
    onChange: onChange,
    iconStart: function iconStart(iconProps) {
      return /*#__PURE__*/React.createElement(SearchIcon, _extends({}, iconProps, {
        size: getSearchIconSize(size)
      }));
    },
    clearIcon: (value === null || value === void 0 ? void 0 : value.length) !== 0 ? /*#__PURE__*/React.createElement(CrossButton, {
      size: size
    }) : undefined,
    clearIconClick: handleClearIconClick
  })));
});
Search.defaultProps = _objectSpread({}, inputDefaultProps);
var iconStart = inputPropTypes.iconStart,
  restInputPropTypes = _objectWithoutProperties(inputPropTypes, ["iconStart"]);
Search.propTypes = restInputPropTypes;
export default Search;
