// @scaleflex/ui/core/time-Picker/time-picker.styles.js
import styled, { css } from 'styled-components';
import IconButton from '../icon-button/icon-button.component';
import { Color as PColor } from '../../utils/types/palette';
import { generateClassNames, applyDisplayNames } from '../../utils/functions';
import inputGroup from '../input-group';
var baseClassName = 'Time-picker';
var TimePicker = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'root')
}).withConfig({
  componentId: "sc-1erl571-0"
})(function (_ref) {
  var fullWidth = _ref.fullWidth;
  return css(["position:relative;width:", ";"], fullWidth ? '100%' : '300px');
});
var TimePickerInput = /*#__PURE__*/styled(inputGroup).attrs({
  className: generateClassNames(baseClassName, 'input')
}).withConfig({
  componentId: "sc-1erl571-1"
})(function (_ref2) {
  var _ref2$isHovering = _ref2.isHovering,
    isHovering = _ref2$isHovering === void 0 ? false : _ref2$isHovering,
    theme = _ref2.theme;
  return css(["input[type='time']::-webkit-calendar-picker-indicator{display:none;-webkit-appearance:none;}", ""], isHovering && css(["border:1px solid ", ";"], theme.palette[PColor.BordersPrimaryHover]));
});
var TimePickerDropdownContainer = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'dropdown-wrapper')
}).withConfig({
  componentId: "sc-1erl571-2"
})(function (_ref3) {
  var theme = _ref3.theme;
  return css(["display:flex;gap:12;max-height:150px;max-width:150px;overflow:auto;cursor:pointer;padding:12px;border-radius:2px;background:", ";box-shadow:0px 2px 6px ", ";"], theme.palette[PColor.BackgroundStateless], theme.palette[PColor.MediumShadow]);
});
var TimePickerDropdown = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'dropdown')
}).withConfig({
  componentId: "sc-1erl571-3"
})(["display:flex;gap:12;max-height:150px;max-width:150px;overflow-x:hidden;cursor:pointer;"]);
var TimePickerDropdownColumn = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'dropdown-column')
}).withConfig({
  componentId: "sc-1erl571-4"
})(function (_ref4) {
  var theme = _ref4.theme;
  return css(["padding:0 6px;max-height:150px;max-width:150px;overflow-y:auto;overflow-x:hidden;cursor:pointer;::-webkit-scrollbar{width:6px;}::-webkit-scrollbar-track{background:#f1f1f1;}::-webkit-scrollbar-thumb{background:", ";border-radius:8px;}"], theme.palette[PColor.BorderPrimaryStateless]);
});
var TimePickerHour = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'hour')
}).withConfig({
  componentId: "sc-1erl571-5"
})(function (_ref5) {
  var theme = _ref5.theme,
    _ref5$selected = _ref5.selected,
    selected = _ref5$selected === void 0 ? false : _ref5$selected;
  return css(["width:30px;height:30px;display:flex;align-items:center;justify-content:center;border-radius:2px;", " &:hover{border:1px solid ", ";box-sizing:border-box;}"], selected && css(["background-color:", ";color:", ";"], theme.palette[PColor.AccentStateless], theme.palette[PColor.IconsInvert]), theme.palette[PColor.AccentPrimaryHover]);
});
var TimePickerMinute = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'minute')
}).withConfig({
  componentId: "sc-1erl571-6"
})(function (_ref6) {
  var theme = _ref6.theme,
    _ref6$selected = _ref6.selected,
    selected = _ref6$selected === void 0 ? false : _ref6$selected;
  return css(["width:30px;height:30px;display:flex;align-items:center;justify-content:center;border-radius:2px;", " &:hover{border:1px solid ", ";box-sizing:border-box;}"], selected && css(["background-color:", ";color:", ";"], theme.palette[PColor.AccentStateless], theme.palette[PColor.IconsInvert]), theme.palette[PColor.AccentPrimaryHover]);
});
var TimePickerPeriod = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'period')
}).withConfig({
  componentId: "sc-1erl571-7"
})(function (_ref7) {
  var theme = _ref7.theme,
    _ref7$selected = _ref7.selected,
    selected = _ref7$selected === void 0 ? false : _ref7$selected;
  return css(["display:flex;align-items:center;justify-content:center;height:30px;width:30px;border-radius:2px;", " &:hover{border:1px solid ", ";box-sizing:border-box;}"], selected && css(["background-color:", ";color:", ";"], theme.palette[PColor.AccentStateless], theme.palette[PColor.IconsInvert]), theme.palette[PColor.AccentPrimaryHover]);
});
var TimePickerIconButton = /*#__PURE__*/styled(IconButton).attrs({
  className: generateClassNames(baseClassName, 'dropdown')
}).withConfig({
  componentId: "sc-1erl571-8"
})(["border:none;width:30px;height:30px;padding:0;"]);
var Styled = applyDisplayNames({
  TimePicker: TimePicker,
  TimePickerInput: TimePickerInput,
  TimePickerDropdownContainer: TimePickerDropdownContainer,
  TimePickerDropdown: TimePickerDropdown,
  TimePickerDropdownColumn: TimePickerDropdownColumn,
  TimePickerHour: TimePickerHour,
  TimePickerMinute: TimePickerMinute,
  TimePickerPeriod: TimePickerPeriod,
  TimePickerIconButton: TimePickerIconButton
});
export default Styled;
