// @scaleflex/ui/core/tag-field/tag-field.styles.js
import styled, { css } from 'styled-components';
import { generateClassNames, applyDisplayNames } from '../../utils/functions';
import { Color as PColor } from '../../utils/types/palette';
import { BorderRadiusSize as BRSize } from '../../utils/types/shape';
import StyledLabel from '../label/label.styles';
import StyledFormHint from '../form-hint/form-hint.styles';
import { sizeTagFieldMixin, fontTagFieldMixin, heightTagFieldMixin, tagFieldBorder } from './tag-field.mixin';
import { Size } from './types';
var baseClassName = 'TagField';
var TagFieldRoot = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'root')
}).withConfig({
  componentId: "sc-10q17fj-0"
})(function (_ref) {
  var fullWidth = _ref.fullWidth;
  return css(["width:", ";", "{margin-top:4px;}", "{margin-bottom:4px;}"], fullWidth ? '100%' : '342px', StyledFormHint.FormHint, StyledLabel.Label);
});
var TagInputFieldWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'tagInputFieldWrapper')
}).withConfig({
  componentId: "sc-10q17fj-1"
})(["position:relative;"]);
var TagFieldWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'tagFieldWrapper')
}).withConfig({
  componentId: "sc-10q17fj-2"
})(function (_ref2) {
  var size = _ref2.size,
    error = _ref2.error,
    theme = _ref2.theme;
  return css(["position:relative;box-sizing:border-box;overflow:hidden;border-radius:4px;background:transparent;display:flex;flex-direction:column;justify-content:space-between;width:100%;", " ", "{margin-top:4px;}", "{margin-bottom:4px;}", " ", ""], tagFieldBorder({
    theme: theme,
    error: error
  }), StyledFormHint.FormHint, StyledLabel.Label, sizeTagFieldMixin[size], heightTagFieldMixin[size]);
});
var TagFieldActions = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'tagFieldActions')
}).withConfig({
  componentId: "sc-10q17fj-3"
})(["display:flex;gap:12px;.SfxButton-Label{font-weight:500;}"]);
var TagFieldLoader = /*#__PURE__*/styled.span.attrs({
  className: generateClassNames(baseClassName, 'loader')
}).withConfig({
  componentId: "sc-10q17fj-4"
})(["display:flex;margin-left:8px;"]);
var TagFieldListWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'listWrapper')
}).withConfig({
  componentId: "sc-10q17fj-5"
})(function (_ref3) {
  var $loading = _ref3.$loading;
  return css(["display:flex;align-items:flex-start;flex-wrap:wrap;gap:8px;margin:0;padding:0;width:100%;overflow:hidden auto;", "{svg{animation:", ";}}"], TagFieldLoader, $loading ? 'spinner 1.2s linear infinite' : '1.2s');
});
var TagFieldInput = /*#__PURE__*/styled.input.attrs({
  className: generateClassNames(baseClassName, 'input')
}).withConfig({
  componentId: "sc-10q17fj-6"
})(function (_ref4) {
  var theme = _ref4.theme;
  return css(["background:transparent;border:none;width:100%;outline:none;color:", ";text-overflow:ellipsis;"], theme.palette[PColor.LinkHover]);
});
var TagFieldInputWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'inputWrapper')
}).withConfig({
  componentId: "sc-10q17fj-7"
})(function (_ref5) {
  var _ref5$size = _ref5.size,
    size = _ref5$size === void 0 ? Size.Md : _ref5$size,
    _ref5$filterInputWidt = _ref5.filterInputWidth,
    filterInputWidth = _ref5$filterInputWidt === void 0 ? 100 : _ref5$filterInputWidt,
    theme = _ref5.theme;
  return css(["background:none;flex-grow:1;height:", "px;display:flex;align-items:center;width:", ";", "{&::-webkit-input-placeholder{color:", ";", "}}"], size === Size.Md ? 26 : 22, Number.isFinite(+filterInputWidth) ? "".concat(filterInputWidth, "px") : filterInputWidth, TagFieldInput, theme.palette[PColor.TextPlaceholder], fontTagFieldMixin[size]);
});
var TagFieldButtonsWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'tagFieldButtonsWrapper')
}).withConfig({
  componentId: "sc-10q17fj-8"
})(function () {
  return css(["display:flex;align-items:center;gap:10px;"]);
});
var TagFieldCopyIcon = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'tagFieldCopyIcon')
}).withConfig({
  componentId: "sc-10q17fj-9"
})(["cursor:pointer;height:16px;margin-left:auto;"]);
var TagFieldSuggestionWrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'suggestionWrapper')
}).withConfig({
  componentId: "sc-10q17fj-10"
})(["position:relative;margin-top:16px;"]);
var TagFieldSuggestionLabel = /*#__PURE__*/styled.label.attrs({
  className: generateClassNames(baseClassName, 'suggestionLabel')
}).withConfig({
  componentId: "sc-10q17fj-11"
})(function (_ref6) {
  var theme = _ref6.theme;
  return css(["display:flex;align-items:center;font-size:12px;padding:6px 0;color:", ";"], theme.palette[PColor.LinkHover]);
});
var TagFieldSuggestionIcon = /*#__PURE__*/styled.span.attrs({
  className: generateClassNames(baseClassName, 'Icon')
}).withConfig({
  componentId: "sc-10q17fj-12"
})(["display:flex;margin-left:4px;"]);
var TagFieldSuggestionWrapperList = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'suggestionWrapperList')
}).withConfig({
  componentId: "sc-10q17fj-13"
})(["display:flex;flex-wrap:wrap;margin:0;padding:0;width:100%;gap:8px;"]);
var TagFieldSuggestionList = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'suggestionList')
}).withConfig({
  componentId: "sc-10q17fj-14"
})(function (_ref7) {
  var theme = _ref7.theme;
  return css(["position:relative;display:flex;align-items:center;margin:0 8px 8px 0;padding:4px 12px;border-radius:", ";border:1px dashed ", ";background:transparent;color:", ";line-height:16.4px;user-select:none;max-height:24px;"], theme.shape.borderRadius[BRSize.Sm], theme.palette[PColor.LinkPrimary], theme.palette[PColor.LinkPrimary]);
});
var Wrapper = /*#__PURE__*/styled.div.attrs({
  className: generateClassNames(baseClassName, 'wrapper')
}).withConfig({
  componentId: "sc-10q17fj-15"
})(["position:relative;"]);
var Styled = applyDisplayNames({
  TagFieldRoot: TagFieldRoot,
  TagInputFieldWrapper: TagInputFieldWrapper,
  TagFieldWrapper: TagFieldWrapper,
  TagFieldActions: TagFieldActions,
  TagFieldListWrapper: TagFieldListWrapper,
  TagFieldInputWrapper: TagFieldInputWrapper,
  TagFieldInput: TagFieldInput,
  TagFieldSuggestionWrapper: TagFieldSuggestionWrapper,
  TagFieldSuggestionLabel: TagFieldSuggestionLabel,
  TagFieldSuggestionIcon: TagFieldSuggestionIcon,
  TagFieldSuggestionWrapperList: TagFieldSuggestionWrapperList,
  TagFieldSuggestionList: TagFieldSuggestionList,
  TagFieldLoader: TagFieldLoader,
  TagFieldButtonsWrapper: TagFieldButtonsWrapper,
  TagFieldCopyIcon: TagFieldCopyIcon,
  Wrapper: Wrapper
});
export default Styled;
